packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -1659,7 +1770,23 @@ export class Combobox
   }
 
   private renderListBoxOptions(): JsxNode {
-    return this.filteredItems.map((item) => (
+    const selectAllComboboxItem = this.selectAllEnabled &&
+      this.selectionMode !== "single" &&
+      this.selectionMode !== "single-persist" && (
+        <calcite-combobox-item
+          ariaLabel={this.messages.selectAll}
+          class={CSS.selectAll}
+          id={`${this.guid}-select-all-enabled`}
+          role="option"
```

Comment:
Why is the role being set here? Shouldn't the component be setting the role? It seems like this would be the responsibility of the component itself.